src-self-hosted/translate_c.zig

```diff
@@ -368,6 +464,424 @@ fn transCompoundStmt(rp: RestorePoint, scope: *Scope, stmt: *const ZigClangCompo
+            const src_type = getExprQualType(c, sub_expr);
+            return TransResult{
+                .node = try transCCast(rp, scope, ZigClangImplicitCastExpr_getBeginLoc(expr), dest_type, src_type, sub_expr_node.node),
+                .node_scope = scope,
+                .child_scope = scope,
+            };
+        },
+        .IntegralCast => {
+            const dest_type = ZigClangExpr_getType(@ptrCast(*const ZigClangExpr, expr));
+            const src_type = ZigClangExpr_getType(sub_expr);
+            return TransResult{
+                .node = try transCCast(rp, scope, ZigClangImplicitCastExpr_getBeginLoc(expr), dest_type, src_type, sub_expr_node.node),
+                .node_scope = scope,
+                .child_scope = scope,
+            };
+        },
+        .FunctionToPointerDecay, .ArrayToPointerDecay => {
+            return maybeSuppressResult(rp, scope, result_used, sub_expr_node);
+        },
+        .LValueToRValue => {
+            return transExpr(rp, scope, sub_expr, .used, .r_value);
+        },
+        else => |kind| return revertAndWarn(
+            rp,
+            error.UnsupportedTranslation,
+            ZigClangStmt_getBeginLoc(@ptrCast(*const ZigClangStmt, expr)),
+            "TODO implement translation of CastKind {}",
+            @tagName(kind),
+        ),
+    }
+}
+
+fn transIntegerLiteral(
+    rp: RestorePoint,
+    scope: *Scope,
+    expr: *const ZigClangIntegerLiteral,
+    result_used: ResultUsed,
+) !TransResult {
+    var eval_result: ZigClangExprEvalResult = undefined;
+    if (!ZigClangIntegerLiteral_EvaluateAsInt(expr, &eval_result, rp.c.clang_context)) {
+        const loc = ZigClangIntegerLiteral_getBeginLoc(expr);
+        return revertAndWarn(rp, error.UnsupportedTranslation, loc, "invalid integer literal");
+    }
+    const node = try transCreateNodeAPInt(rp.c, ZigClangAPValue_getInt(&eval_result.Val));
+    const res = TransResult{
+        .node = node,
+        .child_scope = scope,
+        .node_scope = scope,
+    };
+    return maybeSuppressResult(rp, scope, result_used, res);
+}
+
+fn transReturnStmt(
+    rp: RestorePoint,
+    scope: *Scope,
+    expr: *const ZigClangReturnStmt,
+) !TransResult {
+    const node = try transCreateNodeReturnExpr(rp.c);
+    if (ZigClangReturnStmt_getRetValue(expr)) |val_expr| {
+        const ret_node = node.cast(ast.Node.ControlFlowExpression).?;
+        ret_node.rhs = (try transExpr(rp, scope, val_expr, .used, .r_value)).node;
+    }
+    _ = try appendToken(rp.c, .Semicolon, ";");
+    return TransResult{
+        .node = node,
+        .child_scope = scope,
+        .node_scope = scope,
+    };
+}
+
+fn transStringLiteral(
+    rp: RestorePoint,
+    scope: *Scope,
+    stmt: *const ZigClangStringLiteral,
+    result_used: ResultUsed,
+) !TransResult {
+    const kind = ZigClangStringLiteral_getKind(stmt);
+    switch (kind) {
+        .Ascii, .UTF8 => {
```

Comment:
I think something is wrong with this implementation right now, leading to the garbage output on macOS builds. The approach was to do two passes over the string:
1. get the total length, escaping each character
2. allocate only once based on that size, then fill that buffer with escaped characters

But I probably messed something up here.

Reply:
- `cstr` actually gets the bytes **without** null termination
- no need to call `rp.c.str(cstr)`
- make `cstr` into a slice instead
- rename vars for clarity

```diff
diff --git a/src-self-hosted/translate_c.zig b/src-self-hosted/translate_c.zig
index d6b72f3d..574e395d 100644
--- a/src-self-hosted/translate_c.zig
+++ b/src-self-hosted/translate_c.zig
@@ -687,17 +687,17 @@ fn transStringLiteral(
     const kind = ZigClangStringLiteral_getKind(stmt);
     switch (kind) {
         .Ascii, .UTF8 => {
-            var clen: usize = undefined;
-            const cstr = ZigClangStringLiteral_getString_bytes_begin_size(stmt, &clen);
-            const zstr = try rp.c.str(cstr);
+            var len: usize = undefined;
+            const bytes_ptr = ZigClangStringLiteral_getString_bytes_begin_size(stmt, &len);
+            const str = bytes_ptr[0..len];
 
-            var len: usize = 0;
-            for (zstr) |c| len += escapeChar(c).len;
+            len = 0;
+            for (str) |c| len += escapeChar(c).len;
 
             const buf = try rp.c.a().alloc(u8, len + "c\"\"".len);
             buf[0] = 'c';
             buf[1] = '"';
-            writeEscapedString(buf[2..], zstr);
+            writeEscapedString(buf[2..], str);
             buf[buf.len - 1] = '"';
 
             const token = try appendToken(rp.c, .StringLiteral, buf);
```